src/sema/ast.rs

```diff
@@ -1031,27 +1172,27 @@ impl Recurse for Expression {
                     left.recurse(cx, f);
                     right.recurse(cx, f);
                 }
-                Expression::Subscript(_, _, _, left, right) => {
+                Expression::Subscript {
+                    array: left,
+                    index: right,
+                    ..
+                } => {
                     left.recurse(cx, f);
                     right.recurse(cx, f);
                 }
-                Expression::StructMember(_, _, expr, _) => expr.recurse(cx, f),
+                Expression::StructMember { expr, .. } => expr.recurse(cx, f),
 
-                Expression::AllocDynamicBytes(_, _, expr, _) => expr.recurse(cx, f),
+                Expression::AllocDynamicBytes { length: expr, .. } => expr.recurse(cx, f),
```

Comment:
You can either merge there two branches or use `length` instead of `length: expr` in `Expression::AllocDynamicBytes`.